python/dgl/utils/internal.py

```diff
@@ -784,15 +786,17 @@ def extract_node_subframes(graph, nodes, store_ids=True):
     list[Frame]
         Extracted node frames.
     """
-    if nodes is None:
+    if nodes_or_device is None:
         node_frames = [nf.clone() for nf in graph._node_frames]
-    else:
+    elif isinstance(nodes_or_device, Sequence):
```

Comment:
We currently represent a tensorflow device as a `str` for which `isinstance('/cuda:0', Sequence)` is `True` (which is why CI is failing). We probably need to have a separate parameter for nodes and a separate parameter for device here:
```
def extract_node_subframes(graph, nodes=None, device=None, store_ids=True):
  assert (nodes is None) ^ (device is None), "Only one of nodes or device must be specified."
  ...
```
And then `subgraph.py` would also need to be adjusted to pass them separately.

Reply:
Added an `is_listlike` utility function that returns False for strings instead.